cmd/microcloud/ask.go

```diff
@@ -1104,6 +1104,83 @@ func (c *initConfig) askOVNNetwork(sh *service.Handler) error {
 		}
 	}
 
+	canOVNUnderlay := true
+	for peer, state := range c.state {
+		if len(state.AvailableOVNInterfaces) == 0 {
+			fmt.Printf("Not enough interfaces available on %s to create an underlay network, skipping\n", peer)
+			canOVNUnderlay = false
+			break
+		}
+	}
+
+	var ovnUnderlaySelectedIPs map[string]string
+	if canOVNUnderlay {
+		// TODO: Call `sh.Services[types.MicroOVN].(*service.OVNService).SupportsFeature(context.Background(), "custom_encapsulation_ip")`
+		// when MicroCloud will be updated with microcluster/v2
+
+		wantsDedicatedUnderlay, err := c.asker.AskBool("Configure dedicated underlay networking? (yes/no) [default=no]: ", "no")
+		if err != nil {
+			return err
+		}
+
+		if wantsDedicatedUnderlay {
+			header = []string{"LOCATION", "IFACE", "TYPE", "IP ADDRESS (CIDR)"}
+			fmt.Println("Select exactly one network interface from each cluster member:")
+			data = [][]string{}
+
+			for peer := range c.systems {
+				// skip any systems that have already been clustered, but are available for other configuration.
+				state, ok := c.state[c.name]
+				if ok {
+					if state.ExistingServices[types.MicroOVN][peer] != "" {
+						continue
+					}
+				}
+
+				for _, net := range state.AvailableOVNInterfaces {
+					for _, addr := range net.Addresses {
+						data = append(data, []string{peer, net.Network.Name, net.Network.Type, addr})
+					}
+				}
+			}
+
+			table = NewSelectableTable(header, data)
+			ovnUnderlaySelectedIPs = map[string]string{}
+			err = c.askRetry("Retry selecting underlay network interfaces?", func() error {
+				err = table.Render(table.rows)
+				if err != nil {
+					return err
+				}
+
+				answers, err := table.GetSelections()
+				if err != nil {
+					return err
+				}
+
+				ovnUnderlaySelectedIPs = map[string]string{}
+				for _, answer := range answers {
+					target := table.SelectionValue(answer, "LOCATION")
+					ipAddr := table.SelectionValue(answer, "IP ADDRESS (CIDR)")
+
+					if ovnUnderlaySelectedIPs[target] != "" {
+						return fmt.Errorf("Failed to configure OVN underlay traffic: Selected more than one interface for target %q", target)
+					}
+
+					ovnUnderlaySelectedIPs[target] = ipAddr
+				}
+
+				if len(ovnUnderlaySelectedIPs) != len(c.state) {
```

Comment:
This check is no longer valid, since we can't guarantee all systems will be configured.